qcodes/instrument/abstract.py

```diff
@@ -0,0 +1,101 @@
+from typing import Any, Type
+from qcodes import Parameter
+
+
+class AbstractParameter(Parameter):
+
+    """
+    This is a trivial subclass of 'Parameter' to signal
+    that this parameters *must* be overridden in
+    subclasses of abstract instruments.
+    """
+
+
+class AbstractParameterException(Exception):
+    """
+    Errors with abstract parameter utilization should
+    raise this exception class.
+    """
+
+
+def abstract_instrument(cls: Type) -> Type:
```

Comment:
to make it even better, this could be defined as a type variable to underline that the type of the given class will not change (in other words, that the same class will be returned; or, in yet other words, this decorator does not create a completely new class). don't remember the exact way to do it but along the lines of:
```suggestion
InstrumentBaseT = TypeVar("InstrumentBaseT", bound=Type[InstrumentBase])

def abstract_instrument(cls: InstrumentBaseT) -> InstrumentBaseT:
```